src/Command/Initializer/Profiles/blank/Init.php

```diff
@@ -177,58 +178,37 @@ protected function setupDependencies(string $projectDirectory): int
-            'templates' => [
-                'routes' => [
-                    'sourceFile' => 'app' . DIRECTORY_SEPARATOR . 'return_function.php.template',
-                    'destinationFile' => $bootstrapDirectory . DIRECTORY_SEPARATOR . 'routes.php',
-                    'template' => '',
-                    'replaces' => [],
-                ],
-                'settings' => [
-                    'sourceFile' => 'app' . DIRECTORY_SEPARATOR . 'return_function.php.template',
-                    'destinationFile' => $bootstrapDirectory . DIRECTORY_SEPARATOR . 'settings.php',
-                    'template' => '',
-                    'replaces' => [],
-                ],
-                'dependencies' => [
-                    'sourceFile' => 'app' . DIRECTORY_SEPARATOR . 'return_function.php.template',
-                    'destinationFile' => $bootstrapDirectory . DIRECTORY_SEPARATOR . 'dependencies.php',
-                    'template' => '',
-                    'replaces' => [],
-                ],
-                'middleware' => [
-                    'sourceFile' => 'app' . DIRECTORY_SEPARATOR . 'return_function.php.template',
-                    'destinationFile' => $bootstrapDirectory . DIRECTORY_SEPARATOR . 'middleware.php',
-                    'template' => '',
-                    'replaces' => [],
-                ],
-                'index' => [
-                    'sourceFile' => 'public' . DIRECTORY_SEPARATOR . 'index.php.template',
-                    'destinationFile' => $indexDirectory . DIRECTORY_SEPARATOR . 'index.php',
-                    'template' => '',
-                    'replaces' => [],
-                ],
-            ],
-        ];
+
+        $sourceDirPrefix = $this->templatesDirectory . DIRECTORY_SEPARATOR;
+        $destinationDirPrefix = $directoryFullPath . DIRECTORY_SEPARATOR;
+        $returnFunctionSkeletonFile = $sourceDirPrefix . 'app' . DIRECTORY_SEPARATOR . 'return_function.php.template';
+
+        $routesFileBuilder = new FileBuilder($returnFunctionSkeletonFile);
+        $settingsFileBuilder = new FileBuilder($returnFunctionSkeletonFile);
+        $dependenciesFileBuilder = new FileBuilder($returnFunctionSkeletonFile);
+        $middlewareFileBuilder = new FileBuilder($returnFunctionSkeletonFile);
+        $indexFileBuilder = new FileBuilder($sourceDirPrefix . 'public' . DIRECTORY_SEPARATOR . 'index.php.template');
+
+        $routesBodyReplace = null;
+        $routesPSR7ImportsReplace = null;
+        $settingsImportsReplace = null;
+        $settingsArgumentReplace = null;
+        $settingsBodyReplace = null;
+        $dependenciesImportsReplace = null;
+        $dependenciesArgumentReplace = null;
+        $dependenciesBodyReplace = null;
+        $indexContainerVariableReplace = null;
+        $indexImportsReplace = null;
+        $indexDefineContainerReplace = null;
+        $indexSetContainerReplace = null;
```

Comment:
I would break this entire method into smaller methods instead and call the smaller methods from within this method (`setupDependencies()`).

Let's try to follow single responsibility principle as much as possible:
- `buildRoutesFile()`
- `buildSettingsFile()`
- `buildDependenciesFile()`
- ~~`buildMiddlewareFile()`~~ - This method is unnecessary now as we decided to include all middleware in the main `public/index.php` file
- `buildIndexFile()`

The `FileBuilder` looks good at a preliminary glance.

Also, I see a lot of `<<< BODY >>>` usage which I'm not a big fan of, instead could we extract all of this code within template files instead, read the template then replace the tokens as necessary? It'll much easier to visualize.

Reply:
Sounds better. I'll work on it.